website/addons/s3/model.py

```diff
@@ -5,104 +5,82 @@
 from website.addons.base import StorageAddonBase
 
-from website.addons.s3 import utils
+from website.addons.s3.provider import S3Provider
+from website.addons.s3.serializer import S3Serializer
 from website.addons.s3.settings import ENCRYPT_UPLOADS_DEFAULT
 
 
-class AddonS3UserSettings(AddonUserSettingsBase):
+class S3UserSettings(AddonOAuthUserSettingsBase):
 
-    access_key = fields.StringField()
-    secret_key = fields.StringField()
+    oauth_provider = S3Provider
+    serializer = S3Serializer
 
-    def to_json(self, user):
-        ret = super(AddonS3UserSettings, self).to_json(user)
-        ret['has_auth'] = self.has_auth
-        if self.owner:
-            ret['name'] = self.owner.display_full_name()
-            ret['profile_url'] = self.owner.profile_url
-        return ret
 
-    @property
-    def has_auth(self):
-        return bool(self.access_key and self.secret_key)
-
-    @property
-    def is_valid(self):
-        return utils.can_list(self.access_key, self.secret_key)
-
-    def revoke_auth(self, auth=None, save=False):
-        for node_settings in self.addons3nodesettings__authorized:
-            node_settings.deauthorize(auth=auth, save=True)
-
-        self.s3_osf_user, self.access_key, self.secret_key = None, None, None
-
-        if save:
-            self.save()
-
-        return True
+class S3NodeSettings(StorageAddonBase, AddonOAuthNodeSettingsBase):
 
-
-class AddonS3NodeSettings(StorageAddonBase, AddonNodeSettingsBase):
+    oauth_provider = S3Provider
+    serializer = S3Serializer
 
     bucket = fields.StringField()
     encrypt_uploads = fields.BooleanField(default=ENCRYPT_UPLOADS_DEFAULT)
-    user_settings = fields.ForeignField(
-        'addons3usersettings', backref='authorized'
-    )
 
     @property
     def folder_name(self):
         return self.bucket
 
     @property
+    def folder_id(self):
+        return self.bucket
+
+    @property
+    def folder_path(self):
+        return self.bucket
+
+    def fetch_folder_name(self):
+        return self.folder_name
+
+    @property
     def display_name(self):
         return u'{0}: {1}'.format(self.config.full_name, self.bucket)
 
+    def set_folder(self, folder_id, auth):
+        self.bucket = str(folder_id)
+        self.save()
+
+        self.nodelogger.log(action="bucket_linked", extra={'bucket': folder_id}, save=True)
+
     @property
     def complete(self):
         return self.has_auth and self.bucket is not None
 
     def authorize(self, user_settings, save=False):
         self.user_settings = user_settings
-        self.owner.add_log(
-            action='s3_node_authorized',
-            params={
-                'project': self.owner.parent_id,
-                'node': self.owner._id,
-            },
-            auth=Auth(user_settings.owner),
-        )
-        if save:
-            self.save()
+        self.nodelogger.log(action='node_authorized', save=save)
+
+    def clear_settings(self):
+        self.bucket = None
```

Comment:
Clearing and deauthorizing are completely separate?

IE clearing does not deauthorize and deauthorizing does not clear?